src/eq.rs

```diff
@@ -86,22 +86,22 @@ impl FSSKey for EqKey {
         )
     }
 
-    fn eval(&self, prg: &mut impl PRG, party_id: u8, x: u32) -> u32 {
+    fn eval(&self, prg: &mut impl Prg, party_id: u8, x: u32) -> u32 {
         // Initialize the control bit and the seed.
         assert!((party_id == 0u8) || (party_id == 1u8));
         let mut t_i: u8 = party_id;
         let mut s_i: u128 = self.s;
 
         // Compare the bit decomposition of x with the special path.
         let x_bits: Vec<u8> = bit_decomposition_u32(x);
-        for i in 0..(N * 8) {
+        for (i, x_bit) in x_bits.iter().enumerate().take(N * 8) {
```

Comment:
What warning are you trying to remove? In my opinion the new line is more complicated than the original one. We need to access all the other vectors like `cw[i]` or `t_l[i]` anyway, so I'm not sure it's worth skipping `x[i]`.

Reply:
![image](https://user-images.githubusercontent.com/20444151/134123103-cac63d37-0707-4e17-90c9-7ada01c12f61.png)

Reply:
I did a bit of research and found that we can remove some warnings globally.

We can add:
#![allow(
    clippy::needless_range_loop
)]

to the lib.rs or main.rs. In our case, we can add it to the top of the lib.rs. Do you prefer that?

Reply:
I have reverted the commit with this changes, and add a new one with the ignore code. So that you can see and compare. If you will never care about this warning, I think this is a better option. However, if in the future you want this warning to appear, we have to think on another alternative.

Reply:
Great, I prefer the new version. Tbh this clippy warning does not seem relevant in this context.
Also, readability is really important since the FSS algorithms are already quite complex.